### DISCUSSION TITLE
Code generator warning

### DISCUSSION THREAD

**Discussion by BananaMasterz** [simolus3/drift#3017] [Q&A]
i'm running the run_builder and I get 

```
[WARNING] drift_dev on lib/database/database.dart:
The code generator encountered an issue while attempting to create filters/orderings for $MyTable table. The following filters/orderings were not created: myColumnRefs. Use the @ReferenceName() annotation to resolve this issue.
```

what could be the problem causing this? I've never seen it before (or maybe I haven't noticed and I just ignored it not sure)

**Comment by simolus3** [comment]
This a warning for a new feature. It generates code to [make common queries easier to write](https://drift.simonbinder.eu/docs/manager/) and includes some utilities to write filters on joins easier - that requires resolving references though, and it trips up if there are multiple outgoing references between the same tables.
You can check [this](https://drift.simonbinder.eu/docs/manager/#filtering-across-tables) for an example on how to resolve this (in particular the `Groups` table uses the annotation to avoid the problem). If you don't need the manager APIs, you can also disable `generate_manager` in your [build options](https://drift.simonbinder.eu/docs/advanced-features/builder_options/) to disable everything related to it.

Perhaps the error message should be more actionable and directly suggest adding the annotation to a list of columns or provide a link to more details (cc @dickermoshe).

**Comment by dickermoshe** [comment]
Agree, I should've put more thought into the error message.

This is resolved using the `@ReferenceName()` annotation.
I will add a section to the docs that explains this and I'll also modify the error to be clearer and also link the docs in it.

My Bad
